GSOC 2026 (#1225)
debezium/debezium.github.io: _posts/2026-02-23-gsoc-announce.adoc

---
layout: post
title:  "Debezium Joins Google Summer of Code 2026!"
date:   2026-02-23
tags: [ community, news, gsoc ]
author: gpanice
---

> We’re thrilled to announce that Debezium will participate in Google Summer of Code (GSoC) for 2026! 🎉

Reviewer: ```suggestion
We’re thrilled to announce that Debezium is participating in Google Summer of Code (GSoC) for 2026! 🎉
```

> As part of the https://summerofcode.withgoogle.com/programs/2026/organizations/jboss-community[JBoss Community’s GSoC organization], we’re offering several projects for contributors who want to work on change data capture (CDC) technology and expand the Debezium ecosystem.

Reviewer: ```suggestion
As part of the https://summerofcode.withgoogle.com/programs/2026/organizations/jboss-community[JBoss Community’s GSoC organization], Debezium is offering several projects for contributors to work with change data capture (CDC), vector database technologies, Quarkus, and artificial intelligence (AI) and machine learning (ML).
These are all aimed at expanding the Debezium ecosystem.
```

> GSoC is a fantastic opportunity for aspiring contributors to gain hands‑on engineering experience while being mentored by open-source maintainers. If you’re passionate about distributed systems, data streaming, CDC this is the perfect chance to dive in.

Reviewer: ```suggestion
GSoC is a fantastic opportunity for aspiring contributors to gain hands‑on engineering experience while being mentored by open-source maintainers. If you’re passionate about distributed systems, data streaming, and CDC, this is the perfect opportunity to dive in.
```

== 🚀 Debezium Project Ideas for GSoC 2026

Below are the Debezium‑related projects available for this year’s program, as listed on the official Jboss Community GSoC ideas page:

1. **Debezium Source Connector for Milvus**: implement a new source connector to capture changes from https://milvus.io[Milvus], the popular vector database.

2. **Debezium Source Connector for SQLite** implement a new source connector to capture changes from https://sqlite.org[SQLite], one of the most widely used embedded databases.

3. **Debezium CLI**: a dedicated command-line interface for making setup, debugging, and deployment more intuitive.

4. **Host-Based Pipeline Deployment for the Debezium Platform**: enhance Debezium’s platform-level capabilities by improving the deployment.

5. **PyDebeziumAI**: explore how AI can enhance Debezium via Python

== 💬 Upcoming AMA Session for Contributors

> To help interested candidates get familiar with the projects, mentors, and expectations, we’ll be hosting an Ask‑Me‑Anything (AMA) session. After the session you should be able to begin discussing ideas with mentors.

Reviewer: ```suggestion
To help interested candidates get familiar with the projects, mentors, and expectations, we'll be hosting an Ask-Me-Anything (AMA) session. After the session, candidates should be well-equipped to begin discussing ideas with their mentors.
```

📅 **Date**: 4th March 2026 2.30PM CET

🎬**Platform**: https://www.youtube.com/watch?v=YwdEro2Ji0Q[youtube.com]

🎙️ **Who should attend?** Anyone considering applying for one of the Debezium projects.

== 🤝 How to Get Started

- Read through the full GSoC 2026 https://spaces.redhat.com/spaces/GSOC/pages/750884772/Google+Summer+of+Code+2026+Ideas[idea list] for the JBoss Community.
- Join the discussion official channels for Jboss Community.
- Join the https://debezium.zulipchat.com/#narrow/stream/302529-users[Debezium Zulip Chat] and subscribe to the https://debezium.zulipchat.com/#narrow/channel/573881-community-gsoc[community-gsoc] channel

> - follow Debezium on https://www.linkedin.com/company/debezium[linkedin] for updates

Reviewer: ```suggestion
- Follow Debezium on https://www.linkedin.com/company/debezium[linkedin] for updates
```

- Read the https://github.com/debezium/debezium/blob/main/CONTRIBUTING.md[contribution guidelines]

> - familiarize with Debezium taking a https://github.com/debezium/dbz/issues?q=sort%3Aupdated-desc%20is%3Aissue%20is%3Aopen%20label%3A%22good%20first%20issue%22[good-first-issue]

Reviewer: ```suggestion
- Familiarize yourself with Debezium by taking a https://github.com/debezium/dbz/issues?q=sort%3Aupdated-desc%20is%3Aissue%20is%3Aopen%20label%3A%22good%20first%20issue%22[good-first-issue]
```
